src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Declarations.cs

```diff
@@ -227,21 +291,28 @@ private static IEnumerable<ISymbol> TranslateNamespaces(List<ISymbol> symbols, C
             {
                 var info = await SymbolTreeInfo.GetInfoForAssemblyAsync(solution, assembly, filePath, cancellationToken).ConfigureAwait(false);
 
-                // If the query has a specific string provided, then call into the SymbolTreeInfo
-                // helpers optimized for lookup based on an exact name.
-                if (query.Name != null)
-                {
-                    if (info.HasSymbols(query.Name, query.IgnoreCase))
-                    {
-                        list.AddRange(FilterByCriteria(info.Find(assembly, query.Name, query.IgnoreCase, cancellationToken), filter));
-                    }
-                }
-                else
-                {
+                list.AddRange(FilterByCriteria(Find(query, info, assembly, cancellationToken), filter));
+            }
+        }
+
+        private static IEnumerable<ISymbol> Find(SearchQuery query, SymbolTreeInfo info, IAssemblySymbol assembly, CancellationToken cancellationToken)
+        {
+            // If the query has a specific string provided, then call into the SymbolTreeInfo
+            // helpers optimized for lookup based on an exact name.
+            switch (query.Kind)
+            {
+                case SearchKind.Exact:
+                    return info.Find(assembly, query.Name, ignoreCase: false, cancellationToken: cancellationToken);
+                case SearchKind.ExactIgnoreCase:
+                    return info.Find(assembly, query.Name, ignoreCase: true, cancellationToken: cancellationToken);
+                case SearchKind.Fuzzy:
+                    return info.FuzzyFind(assembly, query.Name, cancellationToken);
+                case SearchKind.Custom:
                     // Otherwise, we'll have to do a slow linear search over all possible symbols.
-                    list.AddRange(FilterByCriteria(info.Find(assembly, query.Predicate, cancellationToken), filter));
-                }
+                    return info.Find(assembly, query.GetPredicate(), cancellationToken);
             }
+
+            throw new InvalidOperationException();
```

Comment:
IMO no.  It should not be possible to create a SearchQuery instance that isn't one of those four values.  So if we didn't hit any of the cases, then we have something invalid internally.  "ArgumentException" is for "i can't take any value from this type domain, but this specific value isn't valid."